Yoast/wordpress-seo
# update Social previews to adapt recent changes

## Context


* As a user of Yoast SEO Premium I want my Facebook and Twitter preview to look just like on Facebook and Twitter.

## Summary


This PR can be summarized in the following changelog entry:

* [wordpress-seo-premium] Updates the social previews to adapt the most recent social media platform changes.

## Relevant technical choices:

- Tools used:
  - debugging the card: [Facebook](https://developers.facebook.com/tools/debug/), [Twitter](https://cards-dev.twitter.com/validator)
  - [Generating images](https://dummyimage.com) to test with

- **Facebook**:
  - [x] make the Title span two lines (landscape)
    - if it spans two lines, don't show the description
  - [x] make the Title and description span max five lines (square and portrait)
    - if the title spans 5 lines, don't show the description
  - [x] increase the font size of the description
  - [x] remove the author
  - [x] update the breakpoints for the different views
    - [x] <200px wide OR <200px high → no image
    - [x] <446 px wide AND/OR <233px high → small image
    - [x] at least 446px wide AND at least 233px high → large image
    - [x] make the image container of the small preview always a square (158x158px) and let the picture always cover that square
    - [x] let the image in the large preview always cover it’s container, like it does in 17.6 and newer
- **Twitter**:
  - [x] move the url to the top
  - [x] remove the link icon before the url
  - [x] update all font colours

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

#### Pre test setup

* Checkout this branch
* In your terminal cd into: `packages/social-metadata-previews/`
* Enter `yarn link`
* Open the `wordpress-seo-premium` repo
* Open the terminal and enter `yarn link "@yoast/social-metadata-previews"`
* Execute the grunt build command `grunt build:js`

#### Test instructions

* Add/edit a post
* Open the Yoast SEO `social` tab
* Add a Facebook and Twitter image and test with different kinds of images
  * Facebook (min width and min height of 200px):
    * Landscape: `width > 446px & height > 223px`
      * Make sure the title can span 2 lines
      * If the title spans 2 lines the description should dissapear
      * The description can span max 1 line 
    * Portrait: `width < height`
    * Square: `width and height are equal`
    * Square:  `width <446px AND/OR height <233px` 
    * Portrait & Square:
      * The title and description together can span max 5 lines
      * If the title spans 4 lines the description can span the remaining 1 etc. 
  * Twitter
    * verify that the url is moved to the top
    * verify that the link icon before the url is removed
    * verify that the font colours are updated 
* Compare the preview images with their card debug/validator tool
    * For Facebook: https://developers.facebook.com/tools/debug/
    * For Twitter: https://cards-dev.twitter.com/validator

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes P1-1049
